_posts/2024-03-25-Announcing-Data-Prepper-2.7.0.md

```diff
@@ -0,0 +1,120 @@
+---
+
+Data Prepper 2.7.0 is now available for [download](https://opensearch.org/downloads.html#data-prepper).
+This release supports extracting geographic locations from IP addresses, supports injectable secrets, and adds many new processors.
+
+
+## GeoIP processor
+
+Data Prepper can now enrich events with geographical location data from an IP address using the new `geoip` processor.
+The `geoip` processor uses the MaxMind GeoLite2 databases to provide geographic location from IP addresses.
+
+Many OpenSearch and Data Prepper users want to enrich their data by adding geographical locations to events.
+There are a number of reasons this data can be valuable.
+Some examples are: looking for anomalies in network access; customer analytics; understanding load across geographies; and more.
+An industry solution for determining a geographical location is through the use of IP addresses.
+
+One example scenario is locating users to a web server.
+Data Prepper already supports parsing Apache Common Log Format for Apache HTTP servers in the `grok` processor.
+In the following example, we can now locate the client making requests using the `clientip` property extracted from the `grok` processor.
+
+```
+processor:
+  - grok:
+      match:
+        log: [ "%{COMMONAPACHELOG_DATATYPED}" ]
+
+  - geoip:
+      entries:
+        - source: clientip
+          target: clientlocation
+          include_fields: [latitude, longitude, location, postal_code, country_name, city_name]
+```
+
+When ingesting data using this pipeline, the OpenSearch index will now contain the geolocation fields expressed above, such as the `latitude` and `city_name`.
+Additionally, you can configure template mappings in OpenSearch so that you can display these events in OpenSearch Dashboards using the [Maps feature](https://opensearch.org/docs/latest/dashboards/visualize/maps/).
+
+
+## AWS Secrets Manager support
+
+Now Data Prepper supports [AWS Secrets Manager](https://aws.amazon.com/secrets-manager/) as extension plugin applicable to pipeline plugins (i.e. source, buffer, processor, sink).
+Users are allowed to configure the AWS secrets extension through extensions in data-prepper-config.yaml.
+
+Here is an example of how to configure your secrets.
+
+```
+extensions:
+  aws:
+    secrets:
+      host-secret-config:
+        secret_id: <YOUR_SECRET_ID_1>
+        region: <YOUR_REGION_1>
+        sts_role_arn: <YOUR_STS_ROLE_ARN_1>
+        refresh_interval: <YOUR_REFRESH_INTERVAL_1>
+      credential-secret-config:
+        secret_id: <YOUR_SECRET_ID_2>
+        region: <YOUR_REGION_2>
+        sts_role_arn: <YOUR_STS_ROLE_ARN_2>
+        refresh_interval: <YOUR_REFRESH_INTERVAL_2>
+```
+
+Users can also configure this in the `pipeline_configurations` section of a pipeline YAML file.
+
+The `credential-secret-config` is a user-supplied secret configuration Id.
```

Comment:
A noun should follow the text in code font.

Reply:
Please see my update and let me know if you have other suggestions.

Reply:
Works for me 😄